src/main/java/gregtech/api/recipes/logic/ParallelLogic.java

```diff
@@ -267,39 +293,73 @@ protected int getMinRatioFluid(Set<FluidStack> countFluid, Recipe recipe, int pa
-        ItemStack itemCopy = itemStack.copy();
-        itemCopy.setCount(count);
-        return itemCopy;
-    }
+    public static RecipeBuilder<?> doParallelRecipes(Recipe currentRecipe, RecipeMap<?> recipeMap, IItemHandlerModifiable importInventory, IMultipleTankHandler importFluids, IItemHandlerModifiable exportInventory, IMultipleTankHandler exportFluids, int parallelAmount) {
+        int multiplierByInputs = getMaxRecipeMultiplier(currentRecipe, importInventory, importFluids, parallelAmount);
+        RecipeBuilder<?> recipeBuilder = recipeMap.recipeBuilder();
+        // Simulate the merging of the maximum amount of recipes
+        // and limit by the amount we can successfully merge
+        int limitByOutput = ParallelLogic.limitByOutputMerging(currentRecipe, exportInventory, exportFluids, multiplierByInputs);
+        int parallelizable = Math.min(multiplierByInputs, limitByOutput);
 
-    protected static FluidStack copyFluidStackWithAmount(FluidStack fluidStack, int count) {
-        FluidStack fluidCopy = fluidStack.copy();
-        fluidCopy.amount = count;
-        return fluidCopy;
+        if (parallelizable > 0) {
+            recipeBuilder.append(currentRecipe, parallelizable);
+        }
+
+        return recipeBuilder;
     }
 
-    protected void multiplyInputsAndOutputs(List<CountableIngredient> newRecipeInputs,
-                                            List<FluidStack> newFluidInputs,
-                                            List<ItemStack> outputItems,
-                                            List<FluidStack> outputFluids,
-                                            Recipe recipe,
-                                            int numberOfOperations) {
+    public static RecipeBuilder<?> appendRecipes(RecipeMap<?> recipeMap, IItemHandlerModifiable importInventory, IMultipleTankHandler importFluids, IItemHandlerModifiable exportInventory, IMultipleTankHandler exportFluids, int parallelAmount, long maxVoltage) {
+        RecipeBuilder<?> recipeBuilder = null;
+
+        OverlayedItemHandler overlayedItemHandler = new OverlayedItemHandler(exportInventory);
+
+        // Iterate over the input items looking for more things to add until we run either out of input items
+        // or we have exceeded the number of items permissible from the smelting bonus
+        int engagedItems = 0;
+
+        for (int index = 0; index < importInventory.getSlots(); index++) {
+            // Skip this slot if it is empty.
+            final ItemStack currentInputItem = importInventory.getStackInSlot(index);
+            if (currentInputItem.isEmpty())
+                continue;
+
+            // Determine if there is a valid recipe for this item. If not, skip it.
+            Recipe matchingRecipe = recipeMap.findRecipe(maxVoltage,
+                    Collections.singletonList(currentInputItem),
+                    Collections.emptyList(), 0, MatchingMode.IGNORE_FLUIDS);
```

Comment:
Since this is no longer Multismelter specific code, we don't want to use `MatchingMode.IGNORE_FLUIDS`, but rather default matching mode